AppController/lib/helperfunctions.rb

```diff
@@ -875,6 +875,14 @@ def self.generate_secure_location_config(handler, port)
     return result
   end
 
+  def self.get_loaded_apps()
```

Comment:
Is checking the filesystem safe since another thread might be updating or setting up that dir?
Isn't zk/db a better place to look for this type of information?